### DISCUSSION TITLE
Modify Card View to show Summary

### DISCUSSION THREAD

**Discussion by afedericojr** [microsoft-search/pnp-modern-search#3748] [Q&A]
The documentation shows how to modify a view template with handlebars for something like the List view, but the Card view doesn't appear to have the same tokenized use in HTML tags, and instead looks to be property driven.

If I add `<div>{{getSummary (slot item @root.slots.Summary)}}</div>` to the card container, since it's not a property, the Summary is called outside the card as unformatted text prior to or after the card, as expected.

```
< div class="template--cardContainer">
            {{#each data.items as |item|}}    
            <!-- IF INSERTED HERE IT SHOWS BEFORE THE CARD -->
                {{#> resultTypes item=item}}
                    <pnp-documentcard class="template--card" 
                        <!-- IF INSERTED HERE IT DOES NOT SHOW -->
                        <!-- IF ADDED HERE AS data-summary="{{getSummary (slot item @root.slots.Summary)}}" IT DOESN'T SHOW -->
                        data-ui-test-id="resultCard" 
                        data-item="{{JSONstringify item}}" 
                        data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.documentCardFields}}" 
                        data-enable-preview="{{@root.properties.layoutProperties.enablePreview}}" 
                        data-show-file-icon="{{@root.properties.layoutProperties.showFileIcon}}" 
                        data-is-compact="{{@root.properties.layoutProperties.isCompact}}" 
                        data-context="{{JSONstringify (truncateContext @root)}}"
                        data-instance-id="{{@root.instanceId}}"
                        data-theme-variant="{{JSONstringify @root.theme}}"
                        data-allow-item-selection="{{@root.properties.itemSelectionProps.allowItemSelection}}"
                        data-item-key="{{@root.paging.currentPageNumber}}{{@index}}"                     
                        data-selected-keys="{{JSONstringify @root.selectedKeys}}"
                        data-index="{{@index}}"
                        >
                    </pnp-documentcard>
                {{/resultTypes}}
             <!-- IF INSERTED HERE IT SHOWS AFTER THE CARD -->
            {{/each}}
```

Can someone please show how to add the hithighlighted summary slot mapping to the Card view template?

Thank you.

**Comment by patrikhellgren** [comment]
The fields used in the Cards layout are static and today there is no way of customizing the layout in pnp-documentcard. You can only modify the contents used for each static field under **Managed card fields**.
You have some different options to accomplish what you want depending of the layout you want with different drawbacks and complexity.

- You could use one of the fields that supports html (Location or Tags) and modify the handlebars template for that with your changes. That way you could add hithighlighted summary and also do some styling and probably even move it around so that it is not locked to the default position of Tags.
- You could copy the html for pnp-documentcard using Devtools in the browser and replace pnp-documentcard in the Cards template and that way you get the html to work with. Then use handlebars to modify the copied html to accomplish the layout you want. You will probably also need to add/modify some styling instead of the dynamic css classes that are added to the html elements in pnp-documentcard.
- You could [develop](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/) your own custom version of the pnp-documentcard web component in a PnP Modern Search Extension. Create an extension solution, copy the source code for the web component from the PnP Modern Search github repo and then make the modifications that you want. Register the extension in the results webpart and then replace the usage of pnp-documentcard with your custom web component.

**Reply by Limerick-gh** [reply]
Worked for me, to inject it into the Tag field. 

After trial and error with the css, it looks quite ok.
![image](https://github.com/user-attachments/assets/29116f67-4740-4827-b4fc-2d866ee89384)
